david_recur2.py

```diff
@@ -0,0 +1,19 @@
+# function returning all permutations of chars of a given string in alphabetical order
+
+def find_all_permutations (input_string):   
```

Comment:
As a general styling rule, you shouldn't have spaces between the function name and parentheses